texar/torch/modules/decoders/decoder_base.py

```diff
@@ -67,9 +68,9 @@ def _make_output_layer(layer: Optional[Union[nn.Module,
                 "Set `output_layer=tx.core.identity` if no output "
                 "layer is wanted.")
         output_layer = nn.Linear(output_size, vocab_size, bias)
-    elif torch.is_tensor(layer):
-        vocab_size = layer.size(0)  # type: ignore
-        output_layer = nn.Linear(layer.size(1),  # type: ignore
+    elif torch.is_tensor(layer) and not callable(layer):
```

Comment:
I don't think this is right -- if `layer` is indeed a callable (and not an `nn.Module`), `ValueError` would be raised. Please fix this and add a test case using a callable as `layer`.

Reply:
Using `Identity()` instead of identity.